### PR TITLE
Add 'set_vertical_balance' method to DSheetPilingModel

### PR DESCRIPTION
**Pull request by KentropDevelopment**
**Fixes issue** 
- #180 

Allows the user to set the parameters needed for checking the vertical balance in D-Sheet Piling. 

It might be good to review if my changes are is consistent and correct regarding the following points:

- In `internal.py` I created `VerticalBalance` and inhereted from `DSeriesInlineMappedProperties`.
- I put the 'user version' of `VerticalBalance` in `constructions.py`, since the vertical balance settings belong to the construction, but not to one element or model specifically.
- I added `test_add_vertical_balance` and gave it the marker unittest.

I am happy to receive feedback on my contribution.

Regards,
Daniël Kentrop

### CODE REVIEW

File: geolib/models/dsheetpiling/constructions.py
Comment (KentropDevelopment) [reply geolib/models/dsheetpiling/constructions.py:5]: I noticed that various objects are present in twofold. One object to be used by the user, possibly having methods. The other for internal use, mainly having attributes containing pydantic typehints. 

An example is the Surface class in surface.py. Surface has a method to_internal which returns the 'internal' Surface object which is defined in internal.py. You can check out other examples in the attached image.

I copied this style to be consistent with the rest of the code. I'd like to here your thoughts on this.

image

**Comment by KentropDevelopment** [comment]
Hi Virginie,

Thanks for reviewing the pull request! I added setting the vertical balance properties to "test_run_sheet_model_acceptance_different_calculation_types" and changed the name of the other test like you suggested.

I'd like to hear your view on the 'internal' point. I can process any suggestions. 

Only somehow the integration tests fail for pydantic v1 python 3.12. The errors occur in code I believe I did not affect. Do you have suggestions for solving this or is this another issue?

Kind regards,
Daniël

**Review by VirginieTrompille — CHANGES_REQUESTED** [review]
I can't approve this pull request if all the tests are not OK. If you fix the merge conflict with the main, it is possible that all the tests will pass.

**Comment by KentropDevelopment** [comment]
The pydantic issue was fixed and I updated my changes accordingly and resolved the merge conflict that emerged in the meanwhile. Only one check fails, with this error:

```
Error output:
No Python at '"C:\hostedtoolcache\windows\Python\3.12.4\x64\python.exe'
```

**Review by VirginieTrompille — APPROVED** [review]

**Pull request merged**
